src/qt_gui/game_list_frame.h

```diff
@@ -18,6 +18,26 @@
 #include "game_list_utils.h"
 #include "gui_context_menus.h"
 
+namespace GameListFrameHelper {
+
+/**
+ * Helper function to perform a case-insensitive comparison of two strings.
+ * This function converts both strings to lowercase and then compares them.
+ * 
+ * @param a First string for comparison.
+ * @param b Second string for comparison.
+ * @return true if a < b, false otherwise.
+ */
+inline bool CompareStringsCaseInsensitive(const std::string& a, const std::string& b) {
```

Comment:
I think having a dedicated namespace for it is a bit overkill, also I think the function is simple enough to not need a detailed comment like this. Also it performs a bunch of extra heap allocations that are unnecessary. The following I didn't test but should have same behavior without any additional overhead
```cpp
bool CompareStringsCaseInsensitive(std::string_view a, std::string_view b) {
    auto lhs = a | std::views::transform(::tolower);
    auto rhs = b | std::views::transform(::tolower);
    return std::ranges::lexicographical_compare(lhs, rhs);
}
```